cmd/server/server.go

```diff
@@ -83,12 +84,8 @@ func main() {
 		log.Fatalf("Cannot create %s store: %v", backend, err)
 	}
 
-	serverOpts := ServerOptions{
-		listeningAddr:     cfg.Server.Host,
-		listeningTLSPort:  cfg.Server.Port,
-		listeningHTTPPort: cfg.Server.HTTPPort,
-	}
-	s := NewServer(serverOpts, kv, cfg.Server.Services)
+	serverOpts := tcprouter.NewServerOptions(cfg.Server.Host, cfg.Server.Port, cfg.Server.HTTPPort, cfg.Server.ClientsPort)
```

Comment:
For option I kinda like the fact that you can just create a literal here. The NewServerOptions doesn't do anything special